aws-sdk-core/lib/aws-sdk-core/plugins/s3_host_id.rb

```diff
@@ -0,0 +1,26 @@
+module Aws
+  module Plugins
+
+    # Support S3 host id, more information, see:
+    # http://docs.aws.amazon.com/AmazonS3/latest/dev/troubleshooting.html#sdk-request-ids
+    #
+    # This plugin adds :host_id for s3 responses when available
+    # @api private
+    class S3HostId < Seahorse::Client::Plugin
+
+      class Handler < Seahorse::Client::Handler
+
+        def call(context)
+          response = @handler.call(context)
+          h = context.http_response.headers
+          context[:host_id] = h['x-amz-id-2']
```

Comment:
Given that this field is S3-specific (especially in case other such fields come up for other services in the future), perhaps a name like `:s3_host_id` would be appropriate.